baseTrie.js

```diff
@@ -99,7 +99,7 @@ Trie.prototype.put = function (key, value, cb) {
       if (self.root.toString('hex') !== ethUtil.SHA3_RLP.toString('hex')) {
         // first try to find the give key or its nearst node
         self.findPath(key, function (err, foundValue, keyRemainder, stack) {
-          if (err) {
+          if (err || !stack) {
             return cb(err)
```

Comment:
If `stack` is undefined, it's likely that `err` is also `undefined` or `null`. It seems this should still call `cb` with an error in this case.

Reply:
If I understood correctly there might be no error, but the `stack` would be undefined. Then `updateNode` would fail because there is no stack.